### PR TITLE
Setting up JsonAccessor class

### PR DESCRIPTION
**Pull request by WillHyde**
The JsonAccessor should allow us to input a url and have it return a string of the JSON at that location

### CODE REVIEW

File: RoomMe/app/src/main/java/com/example/trocket/roomme/AsyncJSONResponse.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.example.trocket.roomme;
+
+/**
+ * Created by Billy on 7/16/15.
+ * AsyncJSONResponse is an interface used to handle a returned JSON string from the JsonAccessor
+ */
+public interface AsyncJSONResponse {
```
Comment (jimmyhlee94) [inline RoomMe/app/src/main/java/com/example/trocket/roomme/AsyncJSONResponse.java:7]: interfaces should be prefixed with a capital I

File: RoomMe/app/src/main/java/com/example/trocket/roomme/HomeActivity.java [hunk 2]
```diff
@@ -58,6 +62,11 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
             }
         });
 
+
+        //Execute the JsonAccessor with an Api address
+        jsonGetter.execute("http://date.jsontest.com/");
```
Comment (jimmyhlee94) [inline RoomMe/app/src/main/java/com/example/trocket/roomme/HomeActivity.java]: comment this line out and denote it as an example

File: RoomMe/app/src/main/java/com/example/trocket/roomme/HomeActivity.java [hunk 3]
```diff
@@ -150,4 +159,13 @@ public boolean onOptionsItemSelected(MenuItem item) {
 
         return super.onOptionsItemSelected(item);
     }
+
+    //onJsonProcessFinish is automatically called when the JsonAccessor finishes
+    //object should be a string that contains the JSON data from the given URL
+    public void onJsonProcessFinish(String object)
```
Comment (jimmyhlee94) [inline RoomMe/app/src/main/java/com/example/trocket/roomme/HomeActivity.java]: this function should return something, not just print to the console.

File: RoomMe/app/src/main/java/com/example/trocket/roomme/JsonAccessor.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                {
+                    builder.append(line);
+                }
+            } else
+            {
+                System.out.println("Failed to Access Json!");
+            }
+        } catch (ClientProtocolException e )
+        {
+            e.printStackTrace();
```
Comment (jimmyhlee94) [inline RoomMe/app/src/main/java/com/example/trocket/roomme/JsonAccessor.java:52]: maybe return a null string or propagate the exception. Right now, the app will just crash

File: RoomMe/app/src/main/java/com/example/trocket/roomme/JsonAccessor.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+        return builder.toString();
+
+    }
+
+    //doInBackground is the method called when a JsonAccessor.execute("url") is called
+    @Override
+    protected String doInBackground(String...urls)
```
Comment (jimmyhlee94) [inline RoomMe/app/src/main/java/com/example/trocket/roomme/JsonAccessor.java:64]: double check that `...urls` is doing what you think it's doing

File: RoomMe/app/src/main/java/com/example/trocket/roomme/JsonAccessor.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    {
+        try{
+            String derp = getJSON(urls[0]);
```
Comment (jimmyhlee94) [inline RoomMe/app/src/main/java/com/example/trocket/roomme/JsonAccessor.java:68]: comment the significance of `urls[0]`

**Comment by jimmyhlee94** [comment]
@zplata you should double check this for me

**Comment by zplata** [comment]
Agree w/ Jimmys comments. Comment the stuff, and fix the little things (his comments), besides the capital I in interface. Otherwise looks fairly simple right now, shouldn't break build or anything.

**Comment by jimmyhlee94** [comment]
:+1:

**Pull request merged**